src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,222 @@
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  GlobalMinimumFrequencyAnnotation
+// |
+// ----------------------------------------------------------------------
+inline GlobalMinimumFrequencyAnnotation::GlobalMinimumFrequencyAnnotation(GlobalMinimumFrequencyAnnotation::FrequencyType value) :
+    Value(std::move(value)) {
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  GlobalMinimumFrequencyEstimator
+// |
+// ----------------------------------------------------------------------
+template <size_t MaxNumTrainingItemsV>
+GlobalMinimumFrequencyEstimator<MaxNumTrainingItemsV>::GlobalMinimumFrequencyEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex) :
+    BaseType(std::move(pAllColumnAnnotations), std::move(colIndex)) {
+}
+
+// ----------------------------------------------------------------------
+// |
+// |  Details::GlobalMinimumFrequencyTrainingOnlyPolicy
+// |
+// ----------------------------------------------------------------------
+template <typename EstimatorT>
+Details::GlobalMinimumFrequencyTrainingOnlyPolicy<EstimatorT>::GlobalMinimumFrequencyTrainingOnlyPolicy(void) :
+    _estimator(
+        GrainedFrequencyEstimatorName,
+        std::make_shared<AnnotationMaps>(
+            [this]() {
+                EstimatorT const & estimator(static_cast<EstimatorT const &>(*this));
+                return estimator.get_column_annotations();
+            }()
+        ),
+        //std::make_shared<AnnotationMaps>(estimator.get_column_annotations()),
+        [](AnnotationMapsPtr pAllColumnAnnotationsParam) {
+            return FrequencyEstimator<>(std::move(pAllColumnAnnotationsParam), 0);
+        }
+    )
+ {
+    _estimator.begin_training();
+}
+
+template <typename EstimatorT>
+void Details::GlobalMinimumFrequencyTrainingOnlyPolicy<EstimatorT>::fit(InputType const &input) {
+    _estimator.fit(input);
+}
+
+template <typename EstimatorT>
+GlobalMinimumFrequencyAnnotation Details::GlobalMinimumFrequencyTrainingOnlyPolicy<EstimatorT>::complete_training(void) {
+    _estimator.complete_training();
+    GrainEstimatorAnnotation<std::string> const & data(GrainedFrequencyEstimator::get_annotation(_estimator.get_column_annotations(), 0, GrainedFrequencyEstimatorName));
+
+    GrainEstimatorAnnotation<std::string>::AnnotationMap annotation(data.Annotations);
+
+    FrequencyType freq(std::chrono::system_clock::duration::max().count());
+    for (auto it = annotation.begin(); it != annotation.end(); it++) {

```

Comment:
>for (auto it = annotation.begin(); it != annotation.end(); it++) { [](start = 4, length = 66)

You don't need to use iterators here, as you only care about the values.

`for(auto const & kvp : data.Annotations)`